modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/discrete/DiscreteNaiveBayesModel.java

```diff
@@ -0,0 +1,124 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.naivebayes.discrete;
+
+import java.io.Serializable;
+import org.apache.ignite.ml.Exportable;
+import org.apache.ignite.ml.Exporter;
+import org.apache.ignite.ml.Model;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+
+/**
+ * Discrete naive Bayes model which predicts result value {@code y} belongs to a class {@code C_k, k in [0..K]} as
+ * {@code p(C_k,y) =x_1*p_k1^x *...*x_i*p_ki^x_i}. Where {@code x_i} is a discrete feature, {@code p_ki} is a prior
+ * probability probability of class {@code p(x|C_k)}. Returns the number of the most possible class.
+ */
+public class DiscreteNaiveBayesModel implements Model<Vector, Double>, Exportable<DiscreteNaiveBayesModel>, Serializable {
+    /** */
+    private static final long serialVersionUID = -127386523291350345L;
+    /**
+     * Probabilities of features for all classes for each label. {@code labels[c][f][b]} contains a probability for
+     * class {@code c} for feature {@code f} for bucket {@code b}.
+     */
+    private final double[][][] probabilities;
+    /** Prior probabilities of each class */
+    private final double[] classProbabilities;
+    /** Labels. */
+    private final double[] labels;
+    /**
+     * The bucket thresholds to convert a features to discrete values. {@code bucketThresholds[f][b]} contains the right
+     * border for feature {@code f} for bucket {@code b}. Everything which is above the last thresdold goes to the next
+     * bucket.
+     */
+    private final double[][] bucketThresholds;
+    /** Amount values in each buckek for each feature per label. */
+    private final DiscreteNaiveBayesSumsHolder sumsHolder;
+
+    /**
+     * @param probabilities Probabilities of features for classes.
+     * @param classProbabilities Prior probabilities for classes.
+     * @param bucketThresholds The threshold to convert a feature to a binary value.
+     * @param sumsHolder Amount values which are abouve the threshold per label.
+     * @param labels Labels.
+     */
+    public DiscreteNaiveBayesModel(double[][][] probabilities, double[] classProbabilities, double[] labels,
+        double[][] bucketThresholds, DiscreteNaiveBayesSumsHolder sumsHolder) {
+        this.probabilities = probabilities;
+        this.classProbabilities = classProbabilities;
+        this.labels = labels;
+        this.bucketThresholds = bucketThresholds;
+        this.sumsHolder = sumsHolder;
+    }
+
+    /** {@inheritDoc} */
+    @Override public <P> void saveModel(Exporter<DiscreteNaiveBayesModel, P> exporter, P path) {
+        exporter.save(this, path);
+    }
+
+    /** Returns a label with max probability. */
+    @Override public Double apply(Vector vector) {
```

Comment:
little comment:
corresponding to our conventions there must be javadoc with "param", "returns" annotations
the same is true for code below (excluding getters and setters as I know)

Reply:
I guess if you write non-empty comment to method then javadoc annotations shuold be added

Reply:
I added Javadoc with annotations.